apps/judicial-system/web/src/api/index.ts

```diff
@@ -1,30 +1,86 @@
 import 'isomorphic-fetch'
-import { Case } from '../types'
+import { Case, CreateCaseRequest } from '../types'
 
-interface GetCasesResponse {
-  ok: boolean
-  code: number
-  cases?: Case[]
-  message?: string
-}
+// const getCaseById: (caseId: string) => Promise<GetCaseByIdResponse> = async (
+//   caseId: string,
+// ) => {
+//   const response = await fetch(`/api/case/${caseId}`)
+
+//   if (response.ok) {
+//     const theCase: Case = await response.json()
+//     return {
+//       httpStatusCode: response.status,
+//       case: theCase,
+//     }
+//   } else {
+//     return {
+//       httpStatusCode: response.status,
+//     }
+//   }
+// }
 
-export const getCases: () => Promise<GetCasesResponse> = async () => {
-  const response = await fetch('/api/cases')
+export const getCases: () => Promise<Case[]> = async () => {
+  try {
+    const response = await fetch('/api/cases')
 
-  if (response.ok) {
-    const cases = await response.json()
-    return {
-      ok: true,
-      code: response.status,
-      cases,
+    if (response.ok) {
+      const cases = await response.json()
+      return cases
+    } else {
+      throw new Error(response.statusText)
     }
-  } else {
+  } catch (ex) {
     // TODO: Log error
+    console.log(ex)
+  }
+}
```

Comment:
Nice error handling

Reply:
Hehh 😅  My thinking is that this is all going to change when we start using GraphQL so I didn't make this my nr.1 priority.

Reply:
Hmm, not sure what I was commenting on here, but remember distinctly that I liked that you were catching all error :)

But we will start with REST, not GraphQL, so at some time we need to do something with the error